### ISSUE TITLE
Menubar: Bad casting from a component.

### ISSUE THREAD

**Issue by josemanuelmartinezsoler** [primefaces/primefaces#8640]
**Describe the defect**
A clear and concise description of what the defect is.

**Environment:**
 - PF Version: 11.0.0_
 - JSF + version: 2.2
 - Affected browsers: ALL

When I call my Model from my bean in my xhtml there should be no problem and get the interface but it gets the DefaultMenuModel and it said it cannot be converted. 

**How I call the menubar**
```xml
<p:menubar model="#{menuMBean.model}"></p:menubar>
```

**My Bean**
```java
@SuppressWarnings("deprecation")
@ManagedBean(name = "menuMBean")
@ViewScoped
public class DynamicMenu extends ControladorMenu{
private MenuModel model;
private List<Menu> listaMenuNivel1;

@Override
@PostConstruct
public void init() {
Service_Factory.getContextoSrv().getLogger().info("Se invoca uno nuevo");

List<Menu> listaMenu;
model = new DefaultMenuModel();

try {
listaMenu = Service_Factory.getMenuSrv().getAll();
listaMenuNivel1 = obtenerNivel1(listaMenu);
construirMenu(listaMenu);
} catch (Exception ex) {
showMessage("msgs", FacesMessage.SEVERITY_ERROR, "Error recuperando usuario logado");
Service_Factory.getContextoSrv().getLogger().error("error", ex);
}
}
```
This is the log what I got. Sorry but is in spanish,

Ha ocurrido un error...
Mensajes: No se puede convertir org.primefaces.model.menu.DefaultMenuModel@35db4468 de tipo class org.primefaces.model.menu.DefaultMenuModel a interface org.primefaces.model.menu.MenuModelStackTrace: javax.el.ELException: No se puede convertir org.primefaces.model.menu.DefaultMenuModel@35db4468 de tipo class org.primefaces.model.menu.DefaultMenuModel a interface org.primefaces.model.menu.MenuModel
at org.apache.el.lang.ELSupport.coerceToType(ELSupport.java:686)
at org.apache.el.ExpressionFactoryImpl.coerceToType(ExpressionFactoryImpl.java:47)
at javax.el.ELContext.convertToType(ELContext.java:306)
at org.apache.el.ValueExpressionImpl.getValue(ValueExpressionImpl.java:186)

Thanks in advance.

**Comment by melloware** [comment]
Fixed in 12.0.0. Duplicate of: https://github.com/primefaces/primefaces/issues/7072

**Comment by josemanuelmartinezsoler** [comment]
Im not setting the ID of the menu so its not the same fix.

**Comment by melloware** [comment]
Please provide an executable example using the [PrimeFaces Test](https://github.com/primefaces/primefaces-test) project.  It is the only way developers can debug your problem to help.  

I can then test if your issue is fixed by the 12.0.0 fix as well.
